### ISSUE TITLE
Adding Multiple Servers to one allocation

### ISSUE THREAD

**Issue by ryan77627** [pterodactyl/panel#1784]
**Describe the solution you'd like**
I mainly use this software as a solution to remote manage my server when I'm out and about. As such, I have one ip: my homes ip. I run a Minecraft server on it's main port (25565), so as such, when I make another, I have to make another allocation (0.0.0.0, 25566) to run the server. Is there any way for me to have multiple servers on the same ip and port so I don't have to swap allocations  for the server? Obviously they can't run at the same time but possibly when attempting to start a server on a port that's already taken have it offer to view the offending server in order to stop it.

**Describe alternatives you've considered**
I mean I can always tell people to either add another server on their client using a different port but if we switch between 5 different servers that's a bit inefficient. Or I could constantly swap the servers allocations to make the current one we're using have the 25565 one, but if I'm having a friend start the server, they won't know how to do that.

**Other notes**
I know this software is meant to be used on big hosting farms, with many ip allocations, for many clients. I'm using this at my house just to remotely manage and check my server. Because of the nature of this, you guys may not want to implement this because the logistics could get ugly real fast (server taking a port and ip that you don't have permission to stop) Do you guys happen to know of any hosting panel that could accomplish what I'm looking for? (I'm not asking you guys to research it, just if you happen to know off the top of your head. Thanks!)

**Comment by parkervcp** [comment]
A port is unique and when in use it cannot be use by any other service when it's already in use. So if people are connecting from outside your house it wouldn't change how this would work.

**Comment by parkervcp** [comment]
This isn't a panel limitation so much as it's a networking limitation.

It's like trying to have 5 phones on one phone number and they are all on different calls.

**Comment by ryan77627** [comment]
Yes I know they cannot be used by more than one service at a time. I just need a way to swap between different servers. As of now I have 3 different servers (different modpacks) on 25565, 25566, and 25567. I'd rather tell my friends just my ip, and run the server on 25565 (again, one at a time). I wouldn't be running more than one at a time on any given port.

**Comment by parkervcp** [comment]
Build a tool to do it with the API. There is no easy way to do this and I doubt any other panel will do it either.

It's not a simple just change 2 values.

You would need to have a secondary unused allocation for each server as you can't have a server without one. Change the default to the secondary on both servers. drop allocations, add allocations, change primary to the new port.

This would still be an issue as the ports are bound to the docker container and each server would need to be rebuilt (started and stopped) to re-bind the ports on each and every change.

**Comment by ryan77627** [comment]
Okay, that makes sense, as I'm not the most familiar with docker and forgot they were run as containers for a bit. Thanks for the help, and I'll keep looking into it to see what the best solution would be for my specific case.

**Comment by SagenKoder** [comment]
> Yes I know they cannot be used by more than one service at a time. I just need a way to swap between different servers. As of now I have 3 different servers (different modpacks) on 25565, 25566, and 25567. I'd rather tell my friends just my ip, and run the server on 25565 (again, one at a time). I wouldn't be running more than one at a time on any given port.

Just for anyone coming here later. Use SRV records on domains to run minecraft servers on different ports but not having to tell your players about the port.

SRV: _minecraft._tcp.{SUBDOMAIN}.{DOMAIN}.{TLD} and point that to the port/domain of the server you want. No need to run every server at 25565.

**Comment by tutacat** [comment]
Or.. you could use proxy forwarding with a config (i.e. nginx (which would need reloading conf) or custom proxy)
Or use a "network"/"universe" minecraft server, that reconnects the user to different servers acting as "dimensions". i.e. PaperMC, Bukkit,

**Comment by xZinio** [comment]
> > Yes I know they cannot be used by more than one service at a time. I just need a way to swap between different servers. As of now I have 3 different servers (different modpacks) on 25565, 25566, and 25567. I'd rather tell my friends just my ip, and run the server on 25565 (again, one at a time). I wouldn't be running more than one at a time on any given port.
> 
> Just for anyone coming here later. Use SRV records on domains to run minecraft servers on different ports but not having to tell your players about the port.
> 
> SRV: _minecraft._tcp.{SUBDOMAIN}.{DOMAIN}.{TLD} and point that to the port/domain of the server you want. No need to run every server at 25565.

Reviving this because I have not found a good solution for my case. Using SRV records with different Minecraft servers running on different ports would mean I would need different domains for each server, right? But really, only one pterodactyl Minecraft server is running on my hardware realistically anyways, so why shouldn't I be able to assign the same allocation to different servers? I have achieved this behavior somewhat by using different IPs that lead to the same destination (e.g. 127.0.0.1:25565 and 0.0.0.0:25565). Yes, this will lead to an address binding error when launching multiple servers at the same time, but as I said, I am not doing that. 

So I am really looking for a solution to use mc.example.org for all my pterodactyl Minecraft servers.  I got this working with three so far (127.0.0.1:25565, 0.0.0.0:25565, 172.18.0.1:25565). But now I am really out of IPs that I can think of that will work. Is there really no way to achieve this in pterodactyl since duplicate allocations are forbidden?
